### PR TITLE
Add recipe for discretizer

### PR DESCRIPTION
**Pull request by basnijholt**
Depends on the following pull requests: 
- [x] `scotch` #1427
- [x] `metis` #1442
- [x] `mumps` #1445
- [x] `tinyarray` #1444
- [x] `kwant` #1446

### CODE REVIEW

File: recipes/discretizer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{% set name = "discretizer" %}
+{% set version = "0.4.3" %}
+{% set sha256 = "de557f087061ea9a0e807e63423d3b325985b262524fdaacdb583749b1eded7a" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: v{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/discretizer/meta.yaml]: Could make this `{{ name }}-{{ version }}.tar.gz`.

File: recipes/discretizer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  url: https://gitlab.kwant-project.org/r-j-skolasinski/discretizer/repository/archive.tar.bz2?ref=v{{ version }}
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+  script: 
+   - python setup.py install --single-version-externally-managed --record record.txt
```
Comment (jakirkham) [inline recipes/discretizer/meta.yaml]: Could use one line instead of using a list.

File: recipes/discretizer/meta.yaml (hunk 1 above)
Comment (basnijholt) [inline recipes/discretizer/meta.yaml]: Note that this is not GitHub, but Gitlab, where this works differently I think.

File: recipes/discretizer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - kwant
+    - numpy
+
+test:
+  imports:
+    - discretizer
+
+about:
+  home: https://gitlab.kwant-project.org/r-j-skolasinski/discretizer
+  license: BSD 2-Clause
```
Comment (basnijholt) [inline recipes/discretizer/meta.yaml]: Done in ee74fa04e9932d519471b6056aa4f3db5fe5da08.

File: recipes/discretizer/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/discretizer/meta.yaml]: So `fn` is just the name the file is saved as by `conda-build` not the name of where it was gotten. Thus it shouldn't matter if it were GitHub, GitLab, or some storage site. Nor should it matter what they chose to name the file.

**Comment by jakirkham** [comment]
Thanks @basnijholt .

Added some more info on `fn` above. That being said, this is a nice to have not an essential thing. So have merged as is.

**Pull request merged**